geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/QueryPlanFilterVisitor.scala

```diff
@@ -316,6 +317,17 @@ class QueryPlanFilterVisitor(sft: SimpleFeatureType) extends DuplicatingFilterVi
     }
   }
 
+  override def visit(filter: PropertyIsLike, extraData: Any): AnyRef = {
+    val pattern = filter.getLiteral
+    try {
+      Pattern.compile(pattern)
```

Comment:
i think we might need to use the LikeToRegexConverter to handle the %'s and all that appropriately: https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/filter/LikeFilterImpl.java#L166
https://github.com/geotools/geotools/blob/e8cf256a9e7ad034863d08a2bc680371f2b774b9/modules/library/main/src/main/java/org/geotools/filter/LikeToRegexConverter.java#L30